modules/programs/vdirsyncer-accounts.nix

```diff
@@ -0,0 +1,284 @@
+        default = null;
+        example = "Password for CalDAV";
+        description = ''
+          Show a prompt for the password with the specified
+          text.
+        '';
+      };
+
+      verify = mkOption {
+        type = types.nullOr types.bool;
+        default = null;
+        description = "Verify SSL certificate. Defaults to <literal>true</literal>.";
+      };
+
+      verifyFingerprint = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        description = ''
+          Optional. SHA1 or MD5 fingerprint of the expected server certificate.</para>
+
+          <para>See
+          <link xlink:href="https://vdirsyncer.pimutils.org/en/stable/ssl-tutorial.html#ssl-tutorial"/>
+          for more information.
+        '';
+      };
+
+      auth = mkOption {
+        type = types.nullOr (types.enum ["basic" "digest" "guess"]);
+        default = null;
+        description = ''
+          Authentication settings. The default is <literal>"basic"</literal>.
+        '';
+      };
+
+      authCert = mkOption {
+        type = types.nullOr (types.either types.str (types.listOf types.str));
+        default = null;
+        description = ''
+        Either a path to a certificate with a client certificate
+        and the key or a list of paths to the files with them.
+       '';
+      };
+
+      userAgent = mkOption {
+        type = types.nullOr types.str;
+        default = null;  
+        description = ''
+          The user agent to report to the server.
+          Defaults to <literal>"vdirsyncer"</literal>.
+        '';
+      };
+
+      ## Options for google storages
+
+      tokenFile = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        description = ''
+          A file path where access tokens
+          are stored.
+        '';
+      };
+
+      clientId = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        description = ''
+          OAuth credentials, obtained from the Google API Manager.</para>
+
+          <para> See
+          <link xlink:href="https://vdirsyncer.pimutils.org/en/stable/config.html#google"/>
+          for more information.
+        '';
+      };
+
+      clientSecret = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        description = ''
+          OAuth credentials, obtained from the Google API Manager.</para>
+
+          <para> See
+          <link xlink:href="https://vdirsyncer.pimutils.org/en/stable/config.html#google"/>
+          for more information.
+        '';
+      };
+    };
+  };
+
+  localModule = types.submodule {
+    options = {
+      type = mkOption {
+        type = types.enum [ "filesystem" "singlefile" ];
+        description = "The type of the storage.";
+      };
+
+      path = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        description = "The path of the storage.";
+      };
+
+      fileExt = mkOption {
```

Comment:
would be nice to be more precise/give some example. It doesn't seem necessary as I don't have one configured.